### ISSUE TITLE
Add Stable Baselines 3 integration for reinforcement learning workloads

### ISSUE THREAD

**Issue by Pendu** [traceopt-ai/traceml#134]
TraceML currently supports PyTorch Lightning and Hugging Face Trainer for supervised learning workloads. It has no integration for reinforcement learning frameworks. Stable Baselines 3 is the most widely used PyTorch RL library and provides a clean `BaseCallback` API that can host TraceML instrumentation.

Add a `TraceMLCallback` for Stable Baselines 3 that instruments RL training runs, separating rollout time (environment interaction) from policy update time (forward/backward/optimizer), with per-layer breakdowns via `trace_model_instance()`.

The callback-based approach means users can enable TraceML with a single line:

```python
from traceml.integrations.stable_baselines3 import TraceMLCallback
model.learn(total_timesteps=100_000, callback=TraceMLCallback())
```

**Scope**

- Add `src/traceml/integrations/stable_baselines3.py` with a `TraceMLCallback(BaseCallback)` class
- Use `_on_training_start` to attach per-layer hooks via `trace_model_instance(self.model.policy)`
- Use `_on_rollout_start` / `_on_rollout_end` boundaries to measure rollout vs. policy-update phase timing
- Respect `TRACEML_DISABLED` environment variable
- Add a minimal example under `examples/sb3_cartpole.py` using PPO + CartPole-v1 (CPU-only, no downloads)
- Add tests under `tests/test_sb3_integration.py`
- Add documentation under `docs/integrations/stable_baselines3.md`

**Design notes**

- SB3's `BaseCallback` does not have hooks around `.train()` (the policy update). The time between `on_rollout_end` and the next `on_rollout_start` captures the update phase by wall-clock difference. Note: for on-policy algorithms, `dump_logs()` runs between these two points (default every iteration), adding I/O overhead to the measurement. The implementation should document this as a known limitation.
- Since `self.model.policy` is a standard `torch.nn.Module`, `trace_model_instance()` provides per-layer forward/backward timing and memory via PyTorch hooks — no SB3-specific code needed for layer-level granularity. The hook target is `self.model.policy` (an `ActorCriticPolicy` or `BasePolicy`), NOT `self.model` itself (which is the algorithm wrapper, not an `nn.Module`).
- **CUDA synchronization:** When timing on GPU (`self.model.device.type == "cuda"`), the callback must call `torch.cuda.synchronize()` before capturing wall-clock timestamps. Without this, CUDA kernel launches are asynchronous and CPU timestamps will severely underestimate GPU execution time.
- **Off-policy step aggregation:** Off-policy algorithms (DQN, SAC, TD3) default to `train_freq=(1, "step")`, meaning callback hooks fire on every single env step. Calling `advance_step()` and `flush_step_events()` per env step will destroy training performance and produce massive telemetry. Implement step aggregation: only advance TraceML steps every N timesteps (e.g., default 1000) for off-policy algorithms.
- **Off-policy warmup:** Off-policy algorithms skip `self.train()` during warmup (`num_timesteps <= learning_starts`). The callback must detect this and not report a policy update phase for warmup iterations.
- **Callback return contract:** `_on_step()` MUST explicitly return `True`. Returning `None` (Python default for missing return) is falsy, which triggers SB3's early-stopping logic and halts training immediately.
- One TraceML "step" should correspond to one complete rollout + policy update cycle (one iteration of the outer `learn()` loop) for on-policy. For off-policy, aggregate N env steps into one TraceML step.
- **Callback ordering:** When combined with other callbacks via `CallbackList`, the execution time of preceding callbacks (e.g., `EvalCallback`, `CheckpointCallback`) in `on_rollout_end` will be included in the "policy update" measurement. Document this and recommend placing `TraceMLCallback` last in the callback list.
- Follow the Lightning integration pattern (`integrations/lightning.py`) for structure and error handling conventions. Wrap all TraceML calls in try/except — SB3 does NOT have fail-open instrumentation, so an unhandled exception in the callback will crash training.

**Acceptance Criteria**

- [ ] Users can enable TraceML with `model.learn(callback=TraceMLCallback())`
- [ ] Callback works with on-policy algorithms (PPO, A2C) and off-policy algorithms (SAC, TD3, DQN)
- [ ] Rollout time and policy update time are reported separately in TraceML's summary
- [ ] Per-layer forward/backward timing and memory appear when running in deep profile mode
- [ ] The callback works on a CPU-only CartPole/Pendulum test without downloading external models or datasets
- [ ] `TRACEML_DISABLED=1` cleanly bypasses all instrumentation
- [ ] No new runtime dependency is added (stable-baselines3 is an optional extra, like lightning/transformers)
- [ ] Errors in TraceML instrumentation never interrupt RL training (fail-open, log to stderr)
- [ ] `_on_step()` explicitly returns `True` (never `None`)
- [ ] CUDA synchronization is performed before timestamp capture when running on GPU
- [ ] Off-policy step aggregation is implemented (not 1 TraceML step per env step)
- [ ] Off-policy warmup phase is detected and no policy-update time is reported during warmup

**References**

- Existing integration patterns: `src/traceml/integrations/lightning.py`, `src/traceml/integrations/huggingface.py`
- Core APIs: `trace_model_instance()` and `trace_step()` in `src/traceml/sdk/instrumentation.py`
- SB3 callback docs: https://stable-baselines3.readthedocs.io/en/master/guide/callbacks.html
- SB3 training loop: `stable_baselines3/common/on_policy_algorithm.py` (on-policy), `stable_baselines3/common/off_policy_algorithm.py` (off-policy)
